src/builder/create_application_command.rs

```diff
@@ -335,10 +311,48 @@ pub struct CreateApplicationCommand {
 }
 
 impl CreateApplicationCommand {
+    /// Create a [`Command`], overriding an existing one with the same name if it exists.
+    ///
+    /// Providing a `command_id` will edit the corresponding command.
+    ///
+    /// Providing a `guild_id` will create a command in the corresponding [`Guild`]. Otherwise, a
+    /// global command will be created.
+    ///
+    /// # Errors
+    ///
+    /// Returns [`Error::Http`] if invalid data is given. See [Discord's docs] for more details.
+    ///
+    /// May also return [`Error::Json`] if there is an error in deserializing the API response.
+    ///
+    /// [Discord's docs]: https://discord.com/developers/docs/interactions/slash-commands
+    #[cfg(feature = "http")]
+    pub async fn execute(
+        self,
+        http: impl AsRef<Http>,
+        guild_id: Option<GuildId>,
+        command_id: Option<CommandId>,
+    ) -> Result<Command> {
+        let http = http.as_ref();
+        match (guild_id, command_id) {
+            (Some(guild_id), Some(command_id)) => {
+                http.edit_guild_application_command(guild_id.into(), command_id.into(), &self).await
+            },
+            (Some(guild_id), None) => {
+                http.create_guild_application_command(guild_id.into(), &self).await
+            },
+            (None, Some(channel_id)) => {
+                http.edit_global_application_command(channel_id.into(), &self).await
+            },
```

Comment:
Typo, commands aren't channels.
```suggestion
            (None, Some(command_id)) => {
                http.edit_global_application_command(command_id.into(), &self).await
            },
```